manual/manual_zx.rst

```diff
@@ -0,0 +1,654 @@
+
+* :py:class:`ZXType.ZXBox`. Similar to the concept of a :py:class:`CircBox` for circuits, a :py:class:`ZXBox` contains another :py:class:`ZXDiagram` abstracted away which can later be expanded in-place. The ports and :py:class:`QuantumType` of incident edges will align with the indices and types of the boundaries on the inner diagram.
+
+Each generator in a diagram is described by a :py:class:`ZXGen` object, or rather an object of one of its concrete subtypes depending on the data needed to describe the generator.
+
+Creating Diagrams
+-----------------
+
+Let's start by making the standard diagram for the qubit teleportation algorithm to showcase the capacity for mixed quantum-classical diagrams. Assuming that the Bell pair will be written in as initialised ancillas rather than open inputs, we just need to start with a diagram with just one quantum input and one quantum output.
+
+.. jupyter-execute::
+
+    import pytket
+    from pytket.zx import ZXDiagram, ZXType, QuantumType, ZXWireType
+    import graphviz as gv
+
+    tele = ZXDiagram(1, 1, 0, 0)
+    (in_v, out_v) = tele.get_boundary()
+    gv.Source(tele.to_graphviz_str())
+
+We will choose to represent the Bell state as a cup (i.e. an edge connecting one side of the CX to the first correction). In terms of vertices, we need two for the CX gate, two for the measurements, and four for the encoding and application of corrections. The CX and corrections need to be coherent operations so will be :py:class:`QuantumType.Quantum` as opposed to the measurements and encodings. We can then link them up by adding edges of the appropriate :py:class:`QuantumType`.
+
+.. jupyter-execute::
+
+    cx_c = tele.add_vertex(ZXType.ZSpider)
+    cx_t = tele.add_vertex(ZXType.XSpider)
+    z_meas = tele.add_vertex(ZXType.ZSpider, qtype=QuantumType.Classical)
+    x_meas = tele.add_vertex(ZXType.XSpider, qtype=QuantumType.Classical)
+    z_enc = tele.add_vertex(ZXType.ZSpider, qtype=QuantumType.Classical)
+    x_enc = tele.add_vertex(ZXType.XSpider, qtype=QuantumType.Classical)
+    z_correct = tele.add_vertex(ZXType.ZSpider)
+    x_correct = tele.add_vertex(ZXType.XSpider)
+
+    # Bell pair between CX and first correction
+    tele.add_wire(cx_t, x_correct)
+
+    # Apply CX between input and first ancilla
+    tele.add_wire(in_v, cx_c)
+    tele.add_wire(cx_c, cx_t)
+
+    # Measure first two qubits
+    tele.add_wire(cx_c, x_meas)
+    tele.add_wire(cx_t, z_meas)
+
+    # Feed measurement outcomes to corrections
+    tele.add_wire(x_meas, x_enc, qtype=QuantumType.Classical)
+    tele.add_wire(x_enc, z_correct)
+    tele.add_wire(z_meas, z_enc, qtype=QuantumType.Classical)
+    tele.add_wire(z_enc, x_correct)
+
+    # Apply corrections to second ancilla
+    tele.add_wire(x_correct, z_correct)
+    tele.add_wire(z_correct, out_v)
+
+    gv.Source(tele.to_graphviz_str())
```

Comment:
Minor thing... Does the difference in line weight denote the difference between quantum and classical wires?

<img width="826" alt="Screenshot 2023-05-26 at 14 18 42" src="https://github.com/CQCL/pytket/assets/93673602/426b66ee-5680-4770-98b4-43310bdbeb9a">